### PR TITLE
add new name-value List

### LINKED ISSUE #1986: NameValueList - Add documentation, examples, guidance to Design System site

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds the new name-value list
#### Where should the reviewer start?
namevaluelist

#### What are the relevant issues?
closes #1986 
#### Should this PR be mentioned in Design System updates?
yes new component
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+>
+  <NameValueListDefaultExample />
+</Example>
+
+## When to use
+
+Consider using a NameValueList for the following use cases:
+
+- Displaying a list of name and value pairs.
+- Presenting metadata in a clear, scannable and consistant way.
```
Comment (taysea) [inline aries-site/src/pages/components/namevaluelist.mdx:43]:
```suggestion
- Presenting metadata in a clear, scannable and consistent way.
```

File: aries-site/src/examples/components/namevaluelist/data.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  Languages: `English, Spanish, French, Chinese, Japanese, German,
+   Korean, Italian, Arabic, Portugese, Russian`,
+  'Operating System':  `VMware ESXi 6.7.0 Build-14320388 Update
+   3 Patch 73 6.7.0 Build-14320388 Update 3 Patch 73 ESXi 6.7.0
+   Build-14320388 Update 3 Patch 73 6.7.0 ESXi 6.7.0 Build-14320388
+   Update 3 Patch 73 6.7.0 `,
+};
+
+export const simpleData = {
+  'Created On': '13/02/2021 at 14:23:32',
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/examples/components/namevaluelist/data.js:57]: honestly Im not sure but I will look at use cases and research if we have any standard date and time

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply aries-site/src/examples/components/namevaluelist/data.js:57]: Any `data` objects should have a date in ISO8601 format. Then, we should have a renderer that uses `Date.toLocaleString()`.

File: aries-site/src/examples/components/namevaluelist/data.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  Policies: '13',
+};
+
+export const anatomyData = {
+  'Single Line Description': 'This is an example of a single-line description',
+  'Multi-line Description': `This is an example of a multi-line description like
+   full-service AI insights, security and unified infrastructure management for
+   campus, branch, remote, and data center networks`,
+  Anchors: (
+    <Anchor label="jane.doe@email.com" href="mailto:jane.doe@email.com" />
```
Comment (taysea) [inline aries-site/src/examples/components/namevaluelist/data.js:18]: Strange but doesn't the anchor look bigger than the rest of the text?

Screen Shot 2021-11-02 at 11 28 39 AM

File: aries-site/src/layouts/content/Example/ExampleResources.js [hunk 4]
```diff
@@ -122,6 +123,11 @@ export const ExampleResources = ({
           </SyntaxHighlighter>
         </CollapsibleSection>
       )}
+      {caption && (
+        <Paragraph size="small" margin="none">
```
Comment (taysea) [inline aries-site/src/layouts/content/Example/ExampleResources.js:127]: I think we should add some bottom margin on the caption (small or medium?) -- haven't tried locally to know what would look better but right now it looks a bit tight

Screen Shot 2021-11-02 at 11 30 00 AM

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+A NameValuePair is a set of two related elements: a name, which identifies a data item, and a value, which is the description of that data.
+
+1. **Name**  
+   A name is a unique indentifier or term referring to a data item. This is typically a constant. In general, name labels are displayed in title case, unless it is a sentence.
+
+2. **Value**
+   Value is the description of data or a pointer to the location of that data. The format of the value can vary, depending on the data set. Content can include: single line description, multi line description, anchors, array or list of items.
+
+  <Example caption="Example: value can include different types of content" pad="none" height={{ min: 'small' }}>
```
Comment (taysea) [inline aries-site/src/pages/components/namevaluelist.mdx:59]:
```suggestion
  <Example caption="Example: Value can include different types of content." pad="none" height={{ min: 'small' }}>
```

File: aries-site/src/layouts/content/Example/ExampleResources.js [hunk 6]
```diff
@@ -122,6 +123,11 @@ export const ExampleResources = ({
...
+          {caption}
+        </Paragraph>
```
Comment (taysea) [inline aries-site/src/layouts/content/Example/ExampleResources.js:129]: Also, for some reason the example with caption isn't aligning with the example below it:

Screen Shot 2021-11-02 at 11 38 00 AM

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+### Font weight
+
+Making the font bolder or thinner is another way to create hierarchy that’s easily identifiable.
+
+- Typically, use a bold weight on names to increase scanability within the list
+- In some cases, a value may be displayed in a bold weight to draw the reader’s attention. An example of this application might be a dashboard page.
+- Don’t use bold text for both name and value at the same time.
+- Don’t use a bold font-weight for any long format content as this can be very hard to read and overwhelm the user.
+
+  <Example caption="Example: A two column Name-Value List where the names are in a bold font-weight" pad="none" height="none">
```
Comment (taysea) [inline aries-site/src/pages/components/namevaluelist.mdx:172]: We should adjust references of "two column" and "multi-column" to match whatever is resolved from the slack poll regarding "Vertical, Horizontal, Grid" etc

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    height="none"
+    bestPractice={{type: 'dont', message: `Never use bold font-weight to display long format content – this can decrease readability.`}}
+  >
+    <NameValueListBoldExampleBad />
+  </Example>
+</BestPracticeGroup>
+
+### Color
+
+Options for text color include the default `text` <ColorSwatch background="text" /> (default for Name-Value pairs) and `text-strong` <ColorSwatch background="text-strong" /> .
```
Comment (taysea) [inline aries-site/src/pages/components/namevaluelist.mdx:215]: Is there any guidance on when `text-strong` would be appropriate?

File: aries-site/src/examples/components/namevaluelist/data.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
+import { Anchor, Box } from 'grommet';
+import { StatusGoodSmall } from 'grommet-icons';
+
+export const alignmentData = {
+  'Created by': (
+    <Anchor label="jane.doe@email.com" href="mailto:jane.doe@email.com" />
```
Comment (taysea) [inline aries-site/src/examples/components/namevaluelist/data.js:6]: Think we should take the approach of having the data structure as:
```
'Created by': 'jane.doe@email.com'
```
Then, in the render of the example, we can say:
```
if (name === 'Created by') return <Anchor label={value} href={`mailto:${value}`} />
```

File: aries-site/src/layouts/content/Example/ExampleResources.js (hunk 4 above)
Comment (britt6612) [reply aries-site/src/layouts/content/Example/ExampleResources.js:127]: I can try this locally to see which one looks good

File: aries-site/src/pages/components/namevaluelist.mdx (hunk 8 above)
Comment (britt6612) [reply aries-site/src/pages/components/namevaluelist.mdx:215]: I can check with @bahriaditi

File: aries-site/src/layouts/content/Example/ExampleResources.js (hunk 6 above)
Comment (britt6612) [reply aries-site/src/layouts/content/Example/ExampleResources.js:129]: I think the 2 above are indented so their smaller I can look into the width

File: aries-site/src/examples/components/namevaluelist/data.js (hunk 3 above)
Comment (britt6612) [reply aries-site/src/examples/components/namevaluelist/data.js:18]: so strange Ill look into this

File: aries-site/src/pages/components/namevaluelist.mdx (hunk 7 above)
Comment (britt6612) [reply aries-site/src/pages/components/namevaluelist.mdx:172]: I'll wait till EOD than implement this

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+<NameValueListAnatomy />
+
+A NameValuePair is a set of two related elements: a name, which identifies a data item, and a value, which is the description of that data.
+
+1. **Name**
+   A name is a unique indentifier or term referring to a data item. This is typically a constant. In general, name labels are displayed in title case, unless it is a sentence.
+
+2. **Value**
+   Value is the description of data or a pointer to the location of that data. The format of the value can vary, depending on the data set. Content can include: single line description, multi line description, anchors, array or list of items.
+
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/pages/components/namevaluelist.mdx:58]: @halocline I just saw this. I had added a `caption` prop to the `Example.js` but I can look at the figure and figcaption

**Comment by britt6612** [comment]
I did explore using `<figure/>` [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption) however there was some weird margin that came with this out of the `box` 
I can look at making a similar component that we can leverage down the line. I still have the work locally so open to how we want to handle this. I did in the meantime add the `caption` prop that will add the text under each example.. [
Screen Shot 2021-11-03 at 10 38 10 PM
](url)

**Pull request merged**